### PR TITLE
doc: use production lists

### PR DESCRIPTION
**Pull request by emillon**
Instead of using code blocks, we use sphinx's built-in support for this.
This looks slightly better but the advantage is that we can use the
following syntax to refer to items:

    :token:`dep`

### CODE REVIEW

File: doc/concepts.rst [hunk 1]
```diff
@@ -66,15 +66,13 @@ Boolean Language
...
-   expr := (and <expr>+)
-         | (or <expr>+)
-         | (<op> <template> <template>)
-         | (not <expr>)
-         | <template>
+.. productionlist::
+   op : '=' | '<' | '>' | '<>' | '>=' | '<='
+   expr : (and `expr`+)
+        : (or `expr`+)
+        : (<op> `template` `template`)
```
Comment (emillon) [inline doc/concepts.rst:73]:
```suggestion
        : (`op` `template` `template`)
```

**Comment by Alizter** [comment]
I think we should avoid using monospace in production lists since it makes the text smaller.

**Comment by emillon** [comment]
> I think we should avoid using monospace in production lists since it makes the text smaller.

to clarify, both are monospace but yes the CSS isn't great when combined with the token syntax.

> duplicate token description of op, other instance in concepts

some namespacing to add, I'll fix that

**Comment by Alizter** [comment]
>to clarify, both are monospace but yes the CSS isn't great when combined with the token syntax.

Yes, I should have clarified. Since they are both mono, let's just remove the backticks.

**Comment by emillon** [comment]
I reverted a plain ASCII syntax on the RHS and fixed the warning.

**Comment by Alizter** [comment]
Looks better!
image

**Review by Alizter — APPROVED** [review]

**Pull request merged**